Cargo.toml

```
@@ -13,10 +13,11 @@ readme = "README.md"
 description = "Grammar framework."
 
 [dependencies]
+cyclotron = "0.0.3"
+elsa = "1.3.2"
 indexmap = "1"
 indexing = "0.3.2"
 proc-macro2 = "0.4.30"
```

```suggestion
proc-macro2 = "1.0.6"
flat-token = "0.0.1"
```
maybe you would need these changes

---

This PR needs rebasing, but I've been too busy lately to finish it.

---

This PR needs to know [what's `gll`. ](https://github.com/rust-lang/gll/issues/130) And how to use it with `grammer`.